jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/telemetry/CodeWhispererCodeCoverageTracker.kt

```diff
@@ -106,9 +138,21 @@ abstract class CodeWhispererCodeCoverageTracker(
     }
 
     private fun emitCodeWhispererCodeContribution() {
-        rangeMarkers.forEach {
-            if (!it.isValid) return@forEach
-            addAndGetAcceptedTokens(it.endOffset - it.startOffset)
+        rangeMarkers.forEach { rangeMarker ->
+            if (!rangeMarker.isValid) return@forEach
+            // if users add more code upon the recommendation generated from CodeWhisperer, we consider those added part as userToken but not CwsprTokens
+            val originalRecommendation = rangeMarker.getUserData(remainingRecommendationKey)
+            val modifiedRecommendation = extractRangeMarkerString(rangeMarker)
+            if (originalRecommendation == null || modifiedRecommendation == null) {
+                LOG.debug {
+                    "failed to get accepted recommendation. " +
+                        "OriginalRecommendation is null: ${originalRecommendation == null}; " +
+                        "ModifiedRecommendation is null: ${modifiedRecommendation == null}"
+                }
+                return
```

Comment:
rangeMarker becomes invalid (all texts are removed in this rangeMaker),modifiedRecommendation will be null here, resulting in an early return. But this is not quite what we want right? If this rangeMarker is invalid, we should continue to other rangeMarkers.

Reply:
yep, when a rangeMarker is invalid, it will return@forEach, so will continue with other rangeMarker instead of early return

Reply:
so this should be return@forEach instead of return right?

Reply:
oh yes, it's my miss, will update